src/repl/scala/tools/nsc/interpreter/IMain.scala

```diff
@@ -817,6 +817,15 @@ class IMain(initialSettings: Settings, protected val out: JPrintWriter) extends
     def value      = Some(handlers.last) filter (h => h.definesValue) map (h => definedSymbols(h.definesTerm.get)) getOrElse NoSymbol
     val lineRep = new ReadEvalPrint()
 
+    private val preambleEndMember = "$ipos"+{lineRep.lineId}
+    // don't look until we've (tried) attributing the trees in unit!
+    lazy val preambleEndPos: Position =
+      unit.body.find {
+        case td: TypeDef => td.name.toString == preambleEndMember
+        case _ => false
+      }.map(_.pos).getOrElse(NoPosition)
```

Comment:
Clever! Another idea is an innocuous import. How about a macro called `$up` or `^` that emits the innocuous marker tree? We just watched our disk of "Up" again, a very effective movie. "Squirrel!"